tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleVersionMojo.java

```diff
@@ -0,0 +1,319 @@
+                    return sb.insert(versionHeaderStartIndex, "$%s&".formatted(bundleVersionHeader)).toString();
+                }
+            }
+        }
+
+        // insert at the end
+        if (wrapProtocolOptions.contains("$")) {
+            // "amp;" is automatically added
+            return sb.insert(localtion.length(), "&%s".formatted(bundleVersionHeader)).toString();
+        } else {
+            return sb.insert(localtion.length(), "$%s".formatted(bundleVersionHeader)).toString();
+        }
+    }
+
+    /**
+     * @param location
+     * @return artifact version first char index, inclusive
+     */
+    int getVersionStartIndex(String location) {
+        char[] chars = location.toCharArray();
+
+        boolean artifactIdFound = false;
+        for (int i = 0; i < chars.length; i++) {
+            if ('/' == chars[i]) {
+                if (!artifactIdFound) {
+                    artifactIdFound = true;
+                } else {
+                    return i + 1;
+                }
+            }
+        }
+
+        return -1;
+    }
+
+    int getVersionEndIndex(String location) {
+        return getVersionEndIndex(location, getVersionStartIndex(location));
+    }
+
+    /**
+     * @param location
+     * @param versionStartIndex
+     * @return artifact version last char index, inclusive
+     */
+    int getVersionEndIndex(String location, int versionStartIndex) {
+        char[] chars = location.toCharArray();
+
+        // start at + 1 to ignore the potential $ coming from version placeholder
+        for (int i = versionStartIndex + 1; i < chars.length; i++) {
+            if ('$' == chars[i]) {
+                return i - 1;
+            }
+        }
+
+        return chars.length - 1;
+    }
+
+    String getVersion(String Location) {
+        return getVersion(Location, getVersionStartIndex(Location), getVersionEndIndex(Location));
+    }
+
+    String getVersion(String location, int versionStartIndex, int versionEndIndex) {
+        return location.substring(versionStartIndex, versionEndIndex + 1);
+    }
+
+    String getValidVersion(String location, String version) throws Exception {
+        if (version.charAt(0) == '$') {
+            throw new Exception("Maven version placeholder '%s' wasn't resolved".formatted(version));
+        }
+
+        try {
+            // Test if version will work in Karaf
+            new Version(version);
+        } catch (Exception e) {
+
+            // TODO: only use cleanVersion if the artifact is non-osgi
+            String cleanVersion = VersionCleaner.clean(version);
+            try {
+                // Test if clean version will work in Karaf again!
+                new Version(cleanVersion);
+
+                getLog().debug(
+                        "Bundle location '%s' will be set with Bundle-Version '%s', the output of org.apache.felix.utils.version.VersionCleaner.clean(%s)"
+                                .formatted(location, cleanVersion, version));
```

Comment:
Consider adding `if (getLog().isDebugEnabled())` for all debug message with non static messages like in this case